server/controllers/finance/reports/operating/index.js

```diff
@@ -0,0 +1,124 @@
+const q = require('q');
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const util = require('../../../../lib/util');
+const Tree = require('../../../../lib/Tree');
+const ReportManager = require('../../../../lib/ReportManager');
+
+const fiscal = require('../../fiscal');
+
+const TEMPLATE = './server/controllers/finance/reports/operating/report.handlebars';
+
+exports.document = document;
+
+function document(req, res, next) {
+  const params = req.query;
+
+  let docReport;
+  const options = _.extend(req.query, {
+    filename : 'TREE.OPERATING_ACCOUNT',
+    csvKey : 'rows',
+    user : req.session.user,
+  });
+
+  try {
+    docReport = new ReportManager(TEMPLATE, req.session, options);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+
+  const queries = [];
+  let range;
+  const EXPENSE_ACCOUNT_TYPE = 5;
+  const INCOME_ACCOUNT_TYPE = 4;
+  const DECIMAL_PRECISION = 2; // ex: 12.4567 => 12.46
+  
+  const getQuery = fiscal.accountBanlanceByTypeId;
+
+  const periods = {
+    periodFrom : params.periodFrom,
+    periodTo : params.periodTo,
+  };
+  
+  fiscal.getDateRangeFromPeriods(periods).then(dateRange => {
+    range = dateRange;
+
+    const totalIncome = `SELECT SUM(r.amount) as total FROM (${getQuery()}) as r`;
```

Comment:
You could actually use the same template here if you wanted.  But this works as well.